src/stores/cardStore.spec.ts

```diff
@@ -0,0 +1,135 @@
+import { CardStore } from "./cardStore";
+import { Card, CardSuit } from "../types";
+
+const fakeCards: Card[] = [
+  { face: "2", suit: CardSuit.Diamond, value: 2, flipped: false },
+  { face: "9", suit: CardSuit.Clover, value: 9, flipped: false },
+  { face: "jack", suit: CardSuit.Clover, value: 0, flipped: false },
+  { face: "3", suit: CardSuit.Heart, value: 3, flipped: false },
+  { face: "4", suit: CardSuit.Heart, value: 4, flipped: false }
+];
+
+describe("create a cardStore with default settings", () => {
+  it("should return cardStore with initial settings", () => {
+    const cardStore = new CardStore();
+
+    expect(cardStore.cards).toHaveLength(288);
+    expect(cardStore.playerCards).toHaveLength(0);
+    expect(cardStore.bankerCards).toHaveLength(0);
+  });
+});
+
+describe("cardStore methods should work", () => {
+  it("should draw a card", () => {
+    const cardStore = new CardStore();
+    const drawnCard = cardStore.drawCard();
+
+    expect(drawnCard).toHaveProperty("face");
+    expect(drawnCard).toHaveProperty("suit");
+    expect(drawnCard).toHaveProperty("value");
+    expect(drawnCard).toHaveProperty("flipped");
+  });
+
+  it("should give player a card", () => {
+    const cardStore = new CardStore();
+    cardStore.givePlayerACard();
+
+    expect(cardStore.playerCards).toHaveLength(1);
+  });
+
+  it("should give banker a card", () => {
+    const cardStore = new CardStore();
+    cardStore.giveBankerACard();
+
+    expect(cardStore.bankerCards).toHaveLength(1);
+  });
+
+  it("should reset player cards", () => {
+    const cardStore = new CardStore();
+    cardStore.resetPlayerCards();
+
+    expect(cardStore.playerCards).toHaveLength(0);
+  });
+
+  it("should reset banker cards", () => {
+    const cardStore = new CardStore();
+    cardStore.resetBankerCards();
+
+    expect(cardStore.bankerCards).toHaveLength(0);
+  });
+});
+
+describe("should count points correctly", () => {
```

Comment:
Technically it can be just
```typescript
it("should count points correctly", () => {
  const cardStore = new CardStore();
  cardStore.playerCards.push(fakeCards[0]); // 2
  cardStore.playerCards.push(fakeCards[1]); // 9
  cardStore.bankerCards.push(fakeCards[2]); // 0
  cardStore.bankerCards.push(fakeCards[3]); // 3

  expect(cardStore.playerPoints).toEqual(1);
  expect(cardStore.bankerPoints).toEqual(3);
});
```

Nested `it()` are useless here